clusterloader2/pkg/measurement/common/scheduler_latency.go

```diff
@@ -208,6 +239,44 @@ func (s *schedulerLatencyMeasurement) getSchedulingLatency(c clientset.Interface
 	return []measurement.Summary{summary}, nil
 }
 
+// Retrieves initial values of scheduler latency
+func (s *schedulerLatencyMeasurement) getSchedulingInitialLatency(c clientset.Interface, host, provider, masterName string, masterRegistered bool) (error) {
+
+	data, err := s.sendRequestToScheduler(c, "GET", host, provider, masterName, masterRegistered)
+	if err != nil {
+		return err
+	}
+	samples, err := measurementutil.ExtractMetricSamples(data)
+	if err != nil {
+		return err
+	}
+
+	s.e2eSchedulingDurationInitHist = measurementutil.NewHistogram(nil)
+	s.schedulingAlgorithmDurationInitHist = measurementutil.NewHistogram(nil)
+	s.preemptionEvaluationInitHist = measurementutil.NewHistogram(nil)
+	s.frameworkExtensionPointDurationInitHist = make(map[string]*measurementutil.Histogram)
+	for _, ePoint := range extentionsPoints {
+		s.frameworkExtensionPointDurationInitHist[ePoint] = measurementutil.NewHistogram(nil)
+	}
+
+	for _, sample := range samples {
```

Comment:
I don't like this code duplication here.

Instead of this whole duplication can you please:

1. introduce a type like "schedulerMetrics" (that contains what you added to schedulerLatencyMeasurement)
2. refactor this method to return (schedulerMetrics, error)
3. call this new function explicitly in getSchedulingLatency() method

Reply:
Will do that.